src/routing/Router.svelte

```diff
@@ -0,0 +1,95 @@
+<script context="module">
+  // We could grab the value of currentQuery here rather than requiring that it
+  // be passed as an argument, but it's unclear how to get an up-to-date store
+  // value from within a module.
+  export function getPath(path, query, preserveQuery = true) {
```

Comment:
What is the use case for `preserveQuery` ? It seems if I don't want to preserve the query, I just wouldn't provide `query`, right?

Reply:
Good call. This was an artifact of me trying to `get` the query within the function's implementation. I'll refactor this.